rtl/eclrtl/rtlcommon.hpp

```diff
@@ -27,14 +27,17 @@ class ECLRTL_API CThorContiguousRowBuffer : implements IRowDeserializerSource
     virtual size32_t readVStr(ARowBuilder & target, size32_t offset, size32_t fixedSize);
     virtual size32_t readVUni(ARowBuilder & target, size32_t offset, size32_t fixedSize);
 
-    //These shouldn't really be called since this class is meant to be used for a deserialize.
-    //If we allowed padding/alignment fields in the input then the first function would make sense.
+    //The following functions should only really be called when used by the readAhead() function
     virtual void skip(size32_t size);
     virtual void skipPackedInt();
     virtual void skipUtf8(size32_t len);
     virtual void skipVStr();
     virtual void skipVUni();
 
+    virtual const byte * querySelf();
```

Comment:
could perhaps be const?

Reply:
Not sure.  The current implementation might read data if nothing has been read yet.